code/game/jobs/whitelist.dm

```diff
@@ -18,35 +18,228 @@ GLOBAL_LIST_FILE_LOAD(whitelist, WHITELISTFILE)
+	list(name = "CO Senator", bitflag = WHITELIST_COMMANDER_LEADER, permission = WL_PANEL_RIGHT_OVERSEER)
+))
+GLOBAL_LIST_INIT(syn_flags, list(
+	list(name = "Synthetic", bitflag = WHITELIST_SYNTHETIC, permission = WL_PANEL_RIGHT_SYNTH),
+	list(name = "Synthetic Council", bitflag = WHITELIST_SYNTHETIC_COUNCIL, permission = WL_PANEL_RIGHT_SYNTH),
+	list(name = "Legacy Synthetic Council", bitflag = WHITELIST_SYNTHETIC_COUNCIL_LEGACY, permission = WL_PANEL_RIGHT_SYNTH),
+	list(name = "Synthetic Senator", bitflag = WHITELIST_SYNTHETIC_LEADER, permission = WL_PANEL_RIGHT_OVERSEER)
+))
+GLOBAL_LIST_INIT(yaut_flags, list(
+	list(name = "Yautja", bitflag = WHITELIST_YAUTJA, permission = WL_PANEL_RIGHT_YAUTJA),
+	list(name = "Legacy Yautja", bitflag = WHITELIST_YAUTJA_LEGACY, permission = WL_PANEL_RIGHT_OVERSEER),
+	list(name = "Yautja Council", bitflag = WHITELIST_YAUTJA_COUNCIL, permission = WL_PANEL_RIGHT_YAUTJA),
+	list(name = "Legacy Yautja Council", bitflag = WHITELIST_YAUTJA_COUNCIL_LEGACY, permission = WL_PANEL_RIGHT_YAUTJA),
+	list(name = "Yautja Senator", bitflag = WHITELIST_YAUTJA_LEADER, permission = WL_PANEL_RIGHT_OVERSEER)
+))
+GLOBAL_LIST_INIT(misc_flags, list(
+	list(name = "Senior Enlisted Advisor", bitflag = WHITELIST_MENTOR, permission = WL_PANEL_RIGHT_MENTOR),
+	list(name = "Working Joe", bitflag = WHITELIST_JOE, permission = WL_PANEL_RIGHT_SYNTH),
+))
+
+/datum/whitelist_panel/ui_static_data(mob/user)
+	. = list()
+	.["co_flags"] = GLOB.co_flags
+	.["syn_flags"] = GLOB.syn_flags
+	.["yaut_flags"] = GLOB.yaut_flags
+	.["misc_flags"] = GLOB.misc_flags
+
+/datum/whitelist_panel/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	var/mob/user = usr
+	if(used_by && (used_by != user.ckey))
+		to_chat(user, SPAN_ALERTWARNING("You are not the current user. [used_by] is editing this player."))
+		return
+	switch(action)
+		if("go_back")
+			if(used_by)
+				used_by = null
+			viewed_player = list()
+			user_rights = 0
+			current_menu = "Panel"
+			target_rights = 0
+			new_rights = 0
+		if("select_player")
+			select_player(user, params["player"])
+			return
+		if("add_player")
+			select_player(user, TRUE)
+			return
+		if("update_number")
+			new_rights = text2num(params["wl_flag"])
+			return
+		if("update_perms")
+			var/player_key = params["player"]
+			var/reason = tgui_input_text(user, "What is the reason for this change?", "Update Reason")
+			if(!reason)
+				return
+			var/datum/entity/player/player = get_player_from_key(player_key)
+			player.set_whitelist_status(new_rights)
+			to_chat(user, SPAN_HELPFUL("Whitelists for [player_key] updated."))
+			message_admins("Whitelists for [player_key] updated by [key_name(user)]. Reason: '[reason]'.")
+			log_admin("WHITELISTS: Flags for [player_key] changed from [target_rights] to [new_rights]. Reason: '[reason]'.")
+			var/datum/entity/player/player2 = get_player_from_key(player_key)
+			target_rights = player2.whitelist_flags
```

Comment:
why do you need to re-get the entity here? `/proc/set_whitelist_status` should happily update the whitelist_flags of the current db entity that you can grab

Reply:
it broke when I tried. No idea why.